src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,158 @@
+
+    .LINK
+    https://aka.ms/ftk/ConvertTo-FinOpsSchema
+#>
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ActualCost,
+
+    [Parameter(Mandatory = $false)]
+    [bool]
+    $ExportAllColumns = $true
+)
+
+function Get-AccountType {
+    [CmdletBinding()]
+    param(
+        [PSCustomObject]
+        $Object
+    )
+
+    $ftk_AccountType = switch ($true) {
+        { $Object.BillingAccountId -eq $Object.BillingProfileId } { "EA" }
+        { $Object.BillingAccountId.Contains(":") } { "MCA" }
+        default { "Other" }
+    }
+    
+    return $ftk_AccountType
+}
+
+
+function ConvertTo-FinOpsSchema {
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+    [CmdletBinding()]
+    param(
+        [string]
+        $ActualCost,
+
+        [bool]
+        $ExportAllColumns
+    )
+
+    if (-not (Test-Path -Path $ActualCost)) {
+        Write-Error "Input file $ActualCost does not exist."
+        return
+    }
+
+    $csvData = Import-Csv -Path $ActualCost
+    
+    $rowCount = $csvData.Count
+    $processedCount = 0
+    $start = [DateTime]::Now
+    
+    $transformedData = $csvData | ForEach-Object {
+        $object = $_
+    
+        # Create a new object with the mapped column names
+        # This will ensure that the output CSV has the correct column names
+        # If exporting all columns, we will use the columnnames mapped here. 
+        $newObject = @{
+            AmortizedCost        = $object.AmortizedCost
+            AvailabilityZone     = $object.AvailabilityZone
+            BilledCost           = $object.BilledCost
+            BillingAccountId     = $object.BillingAccountId
+            BillingAccountName   = $object.BillingAccountName
+            BillingCurrency      = $object.BillingCurrency
+            BillingPeriodEnd     = $object.BillingPeriodEndDate
+            BillingPeriodStart   = $object.BillingPeriodStartDate
+            ChargePeriodEnd      = $object.ChargePeriodEndDate
+            ChargePeriodStart    = $object.ChargePeriodStartDate
+            ChargeType           = $object.ChargeType
+            ProviderName         = $object.ProviderName
+            PublisherName        = $object.PublisherName
+            Region               = $object.Region
+            ResourceId           = $object.ResourceId
+            ResourceName         = $object.ResourceName
+            ServiceCategory      = $object.ServiceCategory
+            ServiceName          = $object.ServiceName
+            SubAccountId         = $object.SubAccountId
+            SubAccountName       = $object.SubAccountName
+            ftk_AccountType      = (Get-AccountType -Object $object)
+            ftk_BillingProfileId = $object.BillingProfileId
+            ftk_BillingProfileName = $object.BillingProfileName
+            ftk_SubscriptionId   = $object.SubscriptionId
```

Comment:
I had to create pseudo code to explain to my devs how to handle this in Cost Management. I converted that to PowerShell. I should be able to do this for all the conversions, but here's EA as a starting point. Can you check to see if it's working correctly?
```suggestion
            AmortizedCost                    = if ($object.IsAmortizedCost) { $object.CostInBillingCurrency } else { 0 }
            AvailabilityZone                 = $object.AvailabilityZone
            BilledCost                       = if ($object.IsActualCost) { $object.CostInBillingCurrency } else { 0 }
            BillingAccountId                 = '/providers/Microsoft.Billing/billingAccounts/' + $object.BillingAccountId
            BillingAccountName               = $object.BillingAccountName
            BillingCurrency                  = $object.BillingCurrencyCode
            BillingPeriodEnd                 = [datetime]::Parse($object.BillingPeriodEndDate).AddDays(1)
            BillingPeriodStart               = [datetime]::Parse($object.BillingPeriodStartDate)
            ChargePeriodStart                = [datetime]::Parse($object.Date)
            ChargePeriodEnd                  = [datetime]::Parse($object.Date).AddDays(1)
            ChargeType                       = if ($object.ChargeType.StartsWith('Unused')) { 'Usage' } elseif (@('Usage', 'Purchase') -contains $object.ChargeType) { $object.ChargeType } else { 'Adjustment' }
            ServiceName                      = $object.ConsumedService # TODO: Convert to ServiceName
            InvoiceIssuerName                = 'Microsoft' # TODO: Get partner name
            ProviderName                     = $object.'Microsoft'
            PublisherName                    = $object.PublisherName
            ResourceId                       = $object.ResourceId
            Region                           = $object.ResourceLocation # TODO: Convert to standard Azure region
            ResourceName                     = $object.ResourceId # TODO: Parse resource name from ResourceId
            SubAccountId                     = '/subscriptions/' + $object.SubscriptionId
            SubAccountName                   = $object.SubscriptionName
            ftk_AccountId                    = $object.AccountId
            ftk_AccountName                  = $object.AccountName
            ftk_AccountOwnerId               = $object.AccountOwnerId
            ftk_AdditionalInfo               = $object.AdditionalInfo
            ftk_EffectivePrice               = $object.EffectivePrice
            ftk_BenefitId                    = $object.benefitId
            ftk_BenefitName                  = $object.benefitName
            ftk_BillingAccountId             = $object.BillingAccountId
            ftk_BillingAccountName           = $object.BillingAccountName
            ftk_BillingAccountType           = if ($accountType -eq 'EA') { 'Billing Account' } elseif ($accountType -eq 'MCA') { 'Billing Profile' } else { 'Subscription' }
            ftk_BillingAccountResourceType   = if ($accountType -eq 'EA') { 'Microsoft.Billing/billingAccounts' } elseif ($accountType -eq 'MCA') { 'Microsoft.Billing/billingAccounts/billingProfiles' } else { 'Microsoft.Resources/subscriptions' }
            ftk_BillingProfileId             = $object.BillingProfileId
            ftk_BillingProfileName           = $object.BillingProfileName
            ftk_ChargeType                   = $object.ChargeType
            ftk_CostAllocationRuleName       = $object.CostAllocationRuleName
            ftk_CostCenter                   = $object.CostCenter
            ftk_CostInPricingCurrency        = $object.CostInBillingCurrency
            ftk_BilledCostInUsd              = if ($object.IsActualCost) { $null } else { 0 } # TODO: Get CostInUsd for EA
            ftk_AmortizedCostInUsd           = if ($object.IsAmortizedCost) { $null } else { 0 } # TODO: Get CostInUsd for EA
            ftk_CustomerName                 = $null
            ftk_CustomerTenantId             = $null
            ftk_ExchangeRatePricingToBilling = if ($accountType -eq 'EA') { 1 } else { $null }
            ftk_ExchangeRateDate             = $object.BillingPeriodStartDate
            ftk_Frequency                    = $object.Frequency
            ftk_InvoiceId                    = $null
            ftk_InvoiceSectionId             = $object.InvoiceSectionId
            ftk_InvoiceSectionName           = $object.InvoiceSectionName
            ftk_IsAzureCreditEligible        = $object.IsAzureCreditEligible
            ftk_MeterCategory                = $object.MeterCategory
            ftk_MeterId                      = $object.MeterId
            ftk_MeterName                    = $object.MeterName
            ftk_MeterRegion                  = $object.MeterRegion
            ftk_MeterSubCategory             = $object.MeterSubCategory
            ftk_OfferId                      = $object.OfferId
            ftk_PartNumber                   = $object.PartNumber
            ftk_PartnerEarnedCreditApplied   = $null
            ftk_PartnerEarnedCreditRate      = $null
            ftk_InvoiceIssuerId              = $null
            ftk_PaygCostInBillingCurrency    = $null # TODO: Get PaygCostInBillingCurrency for EA
            ftk_PaygCostInUsd                = $null # TODO: Get PaygCostInUsd for EA
            ftk_PayGPrice                    = $object.PayGPrice
            ftk_PlanName                     = $object.PlanName
            ftk_PreviousInvoiceId            = $null
            ftk_PricingCurrency              = $object.BillingCurrency
            ftk_PricingModel                 = $object.PricingModel
            ftk_ProductId                    = $object.ProductId # TODO: Confirm this is in the EA dataset (it may be Product)
            ftk_ProductName                  = $object.ProductName
            ftk_ProductOrderId               = $object.ProductOrderId
            ftk_ProductOrderName             = $object.ProductOrderName
            ftk_Provider                     = $null
            ftk_PublisherId                  = $object.PublisherId # TODO: Confirm this is in the EA dataset
            ftk_PublisherType                = $object.PublisherType
            ftk_Quantity                     = $object.Quantity
            ftk_ResellerMpnId                = $null
            ftk_ResellerName                 = $null
            ftk_ResourceGroup                = $object.ResourceGroup
            ftk_ResourceType                 = $object.ResourceId # TODO: Parse type from ResourceId
            ftk_ServiceFamily                = $object.ServiceFamily
            ftk_ServicePeriodEndDate         = [datetime]::Parse($null).AddDays(1)
            ftk_ServicePeriodStartDate       = [datetime]::Parse($null)
            ftk_Tags                         = $object.Tags
            ftk_Term                         = $object.Term
            ftk_UnitOfMeasure                = $object.UnitOfMeasure
            ftk_UnitPrice                    = $object.UnitPrice
```

Reply:
I'll look into it after work and test and make changes if necessary.

Reply:
I committed the changes with some tweaks here and there. Check it out.